src/shims/manual/mscorlib.forwards.cs

```diff
@@ -20,3 +20,4 @@
 [assembly:System.Runtime.CompilerServices.TypeForwardedTo(typeof(System.OrdinalComparer))]
 [assembly:System.Runtime.CompilerServices.TypeForwardedTo(typeof(System.UnitySerializationHolder))]
 [assembly:System.Runtime.CompilerServices.TypeForwardedTo(typeof(System.Diagnostics.Contracts.ContractException))]
+[assembly:System.Runtime.CompilerServices.TypeForwardedTo(typeof(System.Runtime.Serialization.DeserializationBlockedException))]
```

Comment:
This should not be needed. This exception does not exist in desktop, and so there is no point in injecting it into `mscorlib` compat shim. 

Instead, `TypeForwardedFrom` attribute on `DeserializationBlockedException` should forward to `System.Runtime`. It is what new exceptions like `AmbiguousImplementationException` do.

Reply:
It makes me think. This whole feature was designed a speed bump for desktop in the first place. Now that it was rejected for desktop, does the design really make sense for .NET Core?

I think that the binary serialization problem in .NET Core is much smaller and the solution for it can be much more aggressive: Have a switch that disables binary serialization completely (ie replaces it with "throw PNSE"); or that allow us it for managed resources only.

Reply:
This approach would be very similar to how ASP.NET is disallowing synchronous I/O by default in 3.0: https://github.com/aspnet/Announcements/issues/342. If we think that some tech or patterns are bad for you, .NET Core allows us to be clear about it.

Reply:
> This should not be needed. 

That's what I thought (and allConfigurations failed to build for me on the shim), but then tests failed saying it didn't understand the exception.

https://mc.dot.net/#/user/dotnet-bot/pr~2Fdotnet~2Fcorefx~2Frefs~2Fpull~2F34834~2Fmerge/test~2Ffunctional~2Fcli~2F/20190406.35/workItem/System.Runtime.Serialization.Formatters.Tests/analysis/xunit/System.Runtime.Serialization.Formatters.Tests.BinaryFormatterTests~2FValidateAgainstBlobs(obj:%20System.Runtime.Serialization.DeserializationBlockedException:%20message%5CnRemote%20StackTrace%20string)

I'll look into it more and see if I can figure out why it thinks this was needed.

> This whole feature was designed a speed bump for desktop in the first place.

Not entirely.  There are a lot of serializers and a lot of them can be coerced into doing things, this is an additional defensive point for the most common / risky serialization targets.

> Have a switch that disables binary serialization completely

Yeah, that's also being planned.  But managed resources still shouldn't be creating processes while being read, and we eventually want to get all serializers engaging the guard

Reply:
> Not entirely. There are a lot of serializers and a lot of them can be coerced into doing things

I do not think that the complexity of this feature is worth its value in .NET Core. There is a lot of features that poorly designed serializers can expose. Eliminating the tip of the iceberg won't is not moving the needle enough.

> Yeah, that's also being planne

Do we plan to do this for .NET Core 3.0? I think doing this should be much higher priority for .NET Core than this speed bump. I would do this speed bump only after we get some data that it is worth it in the context of .NET Core. (We have data that it is worth it for .NET Framework, but .NET Core is quite different.)